src/routes/api/api-reset-subscription-failed-tasks.ts

```diff
@@ -26,20 +26,27 @@ export const ApiResetSubscriptionFailedTasks = async (req: Request, res: Respons
 
 	const logs: OutputLogRecord[] = [];
 
-	const repoSyncStates = await RepoSyncState.findAllFromSubscription(subscription);
-	await Promise.all(repoSyncStates.map(async (repoSyncState) => {
-		let updated = false;
-		targetTasks.forEach(targetTask => {
-			if (repoSyncState[`${targetTask}Status`] === "failed") {
-				repoSyncState[`${targetTask}Status`] = null;
-				logs.push({ repoSyncStateId: repoSyncState.id, targetTask });
-				updated = true;
+	let offset = 0;
+	let hasNextPage = true;
+	do {
+		const repoSyncStates = await RepoSyncState.findAllFromSubscription(subscription, 100, offset, [["repoFullName", "ASC"]]);
```

Comment:
nit would be better if the 100 value could come from config

Reply:
that's some API endpoint that we probably never use; I wouldn't bother too much about it

Reply:
turned into a constant though :) thanks!